python_modules/dagster/dagster/core/execution.py

```diff
@@ -576,11 +630,26 @@ def _create_resources(pipeline_def, context_def, environment, execution_context,
     # can potentially have many resources so we need to use this abstraction.
     with ExitStack() as stack:
         for resource_name in context_def.resources.keys():
-            resource_obj_or_gen = get_resource_or_gen(
-                pipeline_def, context_def, resource_name, environment, run_id
+
+            resource_def = context_def.resources[resource_name]
+            # Need to do default values
+            resource_config = environment.context.resources.get(resource_name, {}).get('config')
```

Comment:
good catch here. this is a pre-existing bug. we should be apply default values *before* this strongly-typed config object is constructed. can you create an issue and make a comment here referring to that issue?